src/connectionmanager.cpp

```diff
@@ -863,16 +863,15 @@ bool ConnectionManager::switchServers(const ServerData& serverData) {
 void ConnectionManager::backendFailure() {
   logger.error() << "backend failure";
 
-  if (m_state == StateInitializing || m_state == StateOff) {
+  if (m_state == StateInitializing || m_state == StateOff ||
+      m_state == StateOn) {
     emit readyToBackendFailure();
     return;
   }
 
   m_nextStep = BackendFailure;
 
-  if (m_state == StateOn || m_state == StateSwitching ||
-      m_state == StateSilentSwitching || m_state == StateConnecting ||
-      m_state == StateCheckSubscription || m_state == StateConfirming) {
+  if (m_state == StateDisconnecting) {
```

Comment:
Don't we need to deactivate in the case of real backend failures and health check failures, so we transition to a well defined state? For example backend failure is emitted because of failure to launch the tunnel, handshake with VPN server, any other failures in the daemon. Changing the condition here seems risky because we may end up in an undefined state (activated but not working).

Should we instead stop deactivation only on the Guardian heartbeat failure, because that should not affect the VPN?